api/backend.go

```diff
@@ -485,6 +486,52 @@ func (b *StatusBackend) SendDataNotification(dataPayloadJSON string, tokens ...s
 	return err
 }
 
+func (b *StatusBackend) InjectChatAccount(chatKeyHex, encryptionKeyHex string) error {
+	b.mu.Lock()
+	defer b.mu.Unlock()
+
+	b.accountManager.Logout()
+
+	chatKey, err := ethcrypto.HexToECDSA(chatKeyHex)
+	if err != nil {
+		return err
+	}
+
+	err = b.accountManager.InjectChatAccount(chatKey)
+	if err != nil {
+		return err
+	}
+
+	chatAccount, err := b.accountManager.SelectedChatAccount()
+	if err != nil {
+		return err
+	}
+
+	whisperService, err := b.statusNode.WhisperService()
+	switch err {
+	case node.ErrServiceUnknown: // Whisper was never registered
+	case nil:
+		if err := whisperService.SelectKeyPair(chatAccount.AccountKey.PrivateKey); err != nil {
+			return ErrWhisperIdentityInjectionFailure
+		}
+	default:
+		return err
+	}
+
+	if whisperService != nil {
+		st, err := b.statusNode.ShhExtService()
+		if err != nil {
+			return err
+		}
+
+		if err := st.InitProtocol(chatAccount.Address.Hex(), encryptionKeyHex); err != nil {
```

Comment:
Is `InitProtocol` idempotent? What if the protocol is already initialized?

Reply:
@adambabik  it initializes the `shhext`  service, and it's already called from `SelectAccount`, so every time a user logs in